numpy/core/tests/test_api.py

```diff
@@ -342,6 +343,59 @@ def test_copyto():
     # 'dst' must be an array
     assert_raises(TypeError, np.copyto, [1, 2, 3], [2, 3, 4])
 
+def test_copyto_misaligned():
+    # adapted from scipy._lib.tests.test__util.test__aligned_zeros
+    # Checks that unusual memory alignments don't trip up numpy.
+    # In particular, check RELAXED_STRIDES don't trip alignment assertions in
+    # NDEBUG mode for size-0 arrays (gh-12503)
+
+    def aligned_zeros(shape, dtype=float, order="C", align=None):
+        dtype = np.dtype(dtype)
+        if align is None:
+            align = dtype.alignment
+        if not hasattr(shape, '__len__'):
+            shape = (shape,)
+        size = functools.reduce(operator.mul, shape) * dtype.itemsize
```

Comment:
this can be rewritten as np.prod(shape) * dtype.itemsize

Reply:
Gotta be careful of overflow, merged a fix for that just a bit ago that used `dtype=np.intp` in the `prod`, although I might just go with int64.